abltop

tslund opened the pull request:
This is the initial public release of the abltop boundary condition.  I intend to annotate the source files a bit more and do some clean up, but the code is complete enough for your viewing and testing pleasure.  The case ablHill3d was added to the regression test suite and documentation was added to the _Supported Boundary Conditions_ section of the _Theory Manual_.

sayerhs commented on build/do-configNalu_release:
```diff
@@ -21,5 +21,6 @@ cmake \
   -DENABLE_TESTS:BOOL=ON \
   -DENABLE_HYPRE:BOOL=OFF \
   -DHYPRE_DIR:PATH=$hypre_install_dir \
+  -DENABLE_DOCUMENTATION:BOOL=ON \
```
Revert this change

tslund replied:
Done

sayerhs commented on abl_tests/2d_hill_periodic/milestone.xml:
```diff
@@ -0,0 +1,23 @@
+<ParameterList name="MueLu">
```
Do you intend to keep the 2-D tests as part of the regression test suite? If yes, this should move to `reg_tests/test_cases`

tslund replied:
I deleted the 2D test cases

sayerhs commented on reg_tests/test_files/ablHill3d/milestone.xml:
```diff
@@ -0,0 +1,23 @@
+<ParameterList name="MueLu">
```
Use the standard `milestone.xml` from `reg_tests/xml`

tslund replied:
done

sayerhs commented on src/LowMachEquationSystem.C:
```diff
@@ -2985,6 +3074,173 @@ ContinuityEquationSystem::register_symmetry_bc(
+  ScalarFieldType &pressureNone = pressure_->field_of_state(stk::mesh::StateNone);
+  VectorFieldType &dpdxNone = dpdx_->field_of_state(stk::mesh::StateNone);
+
+/*
```
Should this throw an error if the user doesn't select edge based scheme?

tslund replied:
I added some statements here to throw an error for a non edge based scheme.

tslund commented:
I just pushed some changes that address Shreyas' concerns above.  I also added documentation to the AssembleMomentumEdgeABLTopBC.h file and did some general code clean up.   I also added two new test cases to reg_tests.

sayerhs commented:
@tslund Commit e77b5c9ae0c5 adds the mesh file to nalu-wind repo instead of the https://github.com/exawind/meshes submodule repo. Can you please update that before this pull-request is merged?

sayerhs commented:
@michaelasprague This pull request will make FFTW a mandatory dependency of nalu-wind. The build/test infrastructure needs to change for this. Should this be an optional feature that the user enables during CMake configure phase?

jrood-nrel commented:
I am fine with adding FFTW if this will be a fundamental feature of Nalu-Wind, otherwise if it is not, I would prefer to have it optional, but I will defer to @sayerhs and @michaelasprague to decide that.

@tslund I just have one nitpick for this. Would it be possible to make the `find_package` more similar to the other incantations of it in the root `CMakeLists.txt`, and that it reports the FFTW location too?

jrood-nrel commented on CMakeLists.txt:
```diff
@@ -103,6 +115,9 @@ if(ENABLE_TIOGA)
   add_definitions("-DNALU_USES_TIOGA")
 endif()
 
+find_package(FFTW REQUIRED)
```
Thanks for updating the FFTW logic in CMake. These lines just need to be removed, and then I think it looks great.

tslund commented:
Jon,

Thanks for the nod.  I have the code at home on my laptop so I will make
the change that you requested tonight.

I wanted to test my changes, but I ran into difficulty in building nalu.
Perhaps unwisely I merged my branch with the latest Exawind/nalu-wind.
Once I did this, I ran into compilation problems on my macbook pro related
to stk stuff.  It looked to me like trilinos had changed so I updated this
package.  I still ran into compilation problems, related to
trilinos-develop, I think.  I tried to update this package, but ran into
errors when it attemped to build the superlo-dist package.  I feel like I
am diverging on this.  My prior update to trilinos was in late July.  Have
several things changed since then or am I wandering down the wrong path?

   ---Tom

On Tue, Oct 16, 2018 at 10:47 AM Jon Rood <notifications@github.com> wrote:

> *@jrood-nrel* commented on this pull request.
> ------------------------------
>
> In CMakeLists.txt
> <https://github.com/Exawind/nalu-wind/pull/34#discussion_r225621638>:
>
> > @@ -103,6 +115,9 @@ if(ENABLE_TIOGA)
>    add_definitions("-DNALU_USES_TIOGA")
>  endif()
>
> +find_package(FFTW REQUIRED)
>
> Thanks for updating the FFTW logic in CMake. These lines just need to be
> removed, and then I think it looks great.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Exawind/nalu-wind/pull/34#pullrequestreview-165261809>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AgJAFwMgCgdH_p5P4szDzESYa984Iqotks5ulg2ygaJpZM4WbTCL>
> .
>


-- 
Thomas S. Lund
Adjunct Professor and Research Associate
Aerospace Engineering Sciences
University of Colorado, Boulder

jrood-nrel commented:
@tslund Yes, we had to update Trilinos recently for Nalu-Wind. We are currently using a 9-17-2018 `develop` branch version of Trilinos which should match the latest Nalu-Wind, but the latest Trilinos develop should work (though just a few days ago they required CMake >3.10.0). Also, I think superlu-dist should be unnecessary, and superlu is all you should need.

tslund commented:
I think I have the modifications for making FFTW optional working now.  If potential_flow_bc: true appears in the input file beneath abltop_boundary_condition:, then the code must have been compiled with FFTW support ( ENABLE_FFTW set to ON in nalu-wind/CMakeLists.txt).  If not, an error is thrown and the code halts.  abltop_boundary_condition can be used to set normal_temperature_gradient in the absence of potential_flow_bc: true without FFTW support.  I tested these various permutations and everything seemed to work properly.

sayerhs commented:
@tslund Have you updated `exawind/meshes` repo? Also once that is done, we will want to update the `reg_tests/mesh` submodule to point to the latest commit containing the hill mesh from exawind/meshes repo before we merge this.

tslund commented:
I have a topbc branch on my exawind/meshes fork that contains the ablHill3d.exo file.  Should I put this on a pull request?  How do I update the commit pointer on the reg_tests/mesh submodule?

sayerhs commented:
> I have a topbc branch on my exawind/meshes fork that contains the ablHill3d.exo file. Should I put this on a pull request? How do I update the commit pointer on the reg_tests/mesh submodule?

@tslund Since it is just a single mesh update you can push directly to `exawind/meshes` master branch. Pull request is ok too, we can merge it right away. Once the mesh has been committed to the `exawind/meshes` `master` branch, you pull that update from within `reg_tests/mesh` directory and then commit the change in submodule commit SHA ID to the `nalu-wind` and push that update to your `topbc` branch.

tslund commented:
Looks like I do not have permission to push to exawind/meshes directly so I created a pull request.  I guess you need to incorporate the pull request before I can pull the changes back to my nalu-wind?
